README.md

```diff
@@ -1,15 +1,80 @@
-# bde
+# Bayesian Deep Ensembles (BDE)
 
 ## Introduction
 
-The [background paper](https://arxiv.org/abs/2402.01484 )
-and the [repo](https://github.com/EmanuelSommer/bnn_connecting_the_dots) corresponding to the paper.
+This repo was created as a course project at the University of Munich
+(LMU). It implements a bayesian deep ensemble of fully connected 
+networks for use with tabular data.
+The following links contain the [background paper](https://arxiv.org/abs/2402.01484 )
+and the [repo](https://github.com/EmanuelSommer/bnn_connecting_the_dots) corresponding to the background paper. To allow 
+for parallelization, the package is implemented in 
+[Jax](https://jax.readthedocs.io/en/latest/quickstart.html).
 
 ## Development Setup
 
 - Install pre-commit hooks with `pre-commit install`
 - Run tests using `pytest`
 
+## Background
+Bayesian Neural Networks provide a principled approach to deep learning 
+which allows for uncertainty quantification. Compared to traditional
+statistical methods which treat model parameters as unknown, but fixed
+values, Bayesian methods treat model parameters as random
+variables. Hence, we have to specify a prior distribution over those
+parameters which can be interpreted as prior knowledge. Given data,
+we can update the beliefs about the parameters and calculate credible
+intervals for the parameters and predictions. A credible interval in 
+Bayesian statistics defines the range for which the parameter or prediction is 
+believed to fall into with a specified probability based on its posterior distribution. 
+
+However, while potentially rewarding for its predictive capabilities and uncertainty
+measurements, Bayesian optimization can be challenging and resource intensive due to 
+usually strongly multimodal posterior landscapes.
+([Izmailov et al., 2021](https://proceedings.mlr.press/v139/izmailov21a.html))
+To alleviate that issue, this package uses an ensemble of networks sampled from different Markov
+Chains to better capture the posterior density and Jax for computational parallelization.
+
+## The Procedure
+Assumptions: assume an independent distribution of model parameters
+1. define a fully connected neural network structure with two outputs
```

Comment:
The NN can can predict more than 1 distribution, but each predicted distribution should have 2 predicted values.

```suggestion
1. Define a fully connected neural network structure where each output value corresponds to a parametrization of a distribution. 
   In the case of a Gaussian distribution (currently the only supported option), each output value corresponds to 2 predictions: mean $\mu$ and standard deviation $\sigma$, and the output layer for a network with N predicted values should look as follows: $(\mu_1, \mu_2, ..., \mu_N, \sigma_1, \sigma_2, ..., \sigma_N)$.
```